src/core/components/stateful.ts

```diff
@@ -328,9 +330,14 @@ export abstract class StatefulComponentBase extends ComponentBase {
 
         let result = {};
         let state = result[this.globalStateId] = {};
+        let unserializable = [];
         _.forOwn(this.__stateMetadata, (metadata, key) => {
             let value = this[metadata.propertyName];
 
+            if (!metadata.shared && !isJSONSerializable(value)) {
+                unserializable.push(`[global state id: ${this.globalStateId}] ${metadata.propertyName}: ${value}`);
```

Comment:
The output of only `${value}` here will not be very useful for objects :/
Only if value is directly a function or undefined, it will stringify it nicely

Reply:
Maybe:
```ts
const verboseValue = JSON.stringify(value, (key, value) => {
    if (typeof value === 'function' ) return value.toString();
    if (typeof value === 'undefined') return 'undefined';
    if (typeof value === 'number' && value === Infinity) return 'Infinity';
    if (typeof value === 'number' && value === -Infinity) return '-Infinity';
    if (typeof value === 'number' && isNaN(value)) return 'NaN';
    return value;
});

`[global state id: ${this.globalStateId}] ${metadata.propertyName}: ${value} ${verboseValue}`
```